llvm/unittests/Demangle/PartialDemangleTest.cpp

```diff
@@ -16,45 +16,70 @@ struct ChoppedName {
 };
 
 static ChoppedName NamesToTest[] = {
-  {"_Z1fv", "", "f", "", "()"},
-  {"_ZN1a1b1cIiiiEEvm", "a::b", "c", "void", "(unsigned long)"},
-  {"_ZZ5OuterIiEivEN5Inner12inner_memberEv",
-   "int Outer<int>()::Inner", "inner_member", "", "()"},
-  {"_Z1fIiEPFvvEv", "", "f", "void (*)()", "()"},
-  {"_ZN1S1fIiEEvv", "S", "f", "void", "()"},
-
-  // Call operator for a lambda in f().
-  {"_ZZ1fvENK3$_0clEi", "f()::$_0", "operator()", "", "(int)"},
-
-  // A call operator for a lambda in a lambda in f().
-  {"_ZZZ1fvENK3$_0clEvENKUlvE_clEv",
-   "f()::$_0::operator()() const::'lambda'()", "operator()", "", "()"},
-
-  {"_ZZN1S1fEiiEd0_NKUlvE_clEv",
-   "S::f(int, int)::'lambda'()", "operator()", "", "()"},
-
-  {"_ZN1Scv7MuncherIJDpPT_EEIJFivEA_iEEEv",
-   "S", "operator Muncher<int (*)(), int (*) []>", "", "()"},
-
-  // Attributes.
-  {"_ZN5test4IdE1fEUa9enable_ifIXeqfL0p_Li1EEXeqfL0p0_Li2EEEi",
-   "test4<double>", "f", "", "(int)"},
-  {"_ZN1SC2B8ctor_tagEv", "S", "S", "", "()"},
-  {"_ZN1S1fB4MERPIiEEvv", "S", "f", "void", "()"},
-
-  {"_ZNSsC1EmcRKSaIcE",
-   "std::basic_string<char, std::char_traits<char>, std::allocator<char>>",
-   "basic_string", "", "(unsigned long, char, std::allocator<char> const&)"},
-  {"_ZNSsixEm", "std::string", "operator[]", "", "(unsigned long)"},
-  {"_ZSt17__throw_bad_allocv", "std", "__throw_bad_alloc", "", "()"},
-
-  {"_ZN1AI1BEC2Ev", "A<B>", "A", "", "()"},
-  {"_ZN1AI1BED2Ev", "A<B>", "~A", "", "()"},
-  {"_ZN1AI1BECI24BaseEi", "A<B>", "A", "", "(int)"},
-  {"_ZNKR1AI1BE1fIiEEiv", "A<B>", "f", "int", "()"},
-
-  {"_ZN1SIJicfEE3mfnIJjcdEEEvicfDpT_", "S<int, char, float>",
-   "mfn", "void", "(int, char, float, unsigned int, char, double)"},
+    {"_Z1fv", "", "f", "", "()"},
```

Comment:
Tests should be added in `libcxxabi/test/test_demangle.pass.cpp` as well: the libc++abi copy of the demangler is the canonical one.